libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/SpeechAudioFocusManager.java

```diff
@@ -0,0 +1,45 @@
+package com.mapbox.services.android.navigation.ui.v5.voice;
+
+import android.content.Context;
+import android.media.AudioFocusRequest;
+import android.media.AudioManager;
+import android.os.Build;
+
+class SpeechAudioFocusManager {
+
+  private AudioManager audioManager;
+  private AudioFocusRequest audioFocusRequest;
+
+  SpeechAudioFocusManager(Context context) {
+    initAudioManager(context);
+    initAudioFocusRequest();
+  }
+
+  void requestAudioFocus() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
Could we remove/abstract these kind of checks somehow? I'm thinking on implementing like a mix of [_Factory_](https://sourcemaking.com/design_patterns/factory_method) and [_Strategy_](https://sourcemaking.com/design_patterns/strategy) patterns.